File: 0041-first-missing-positive/0041-first-missing-positive.cpp
class Solution {
public:
    int firstMissingPositive(vector<int>& a) {
        unordered_map<int,int>map;
        int i,j,k,c=0;

        for(auto x:a){
            if(x>0)
            map[x]++;
        }

        for(i=1;i<a.size()+1;i++){
            if(map.find(i)==map.end())
            return i;

        }
        return a.size()+1;
        

           }
};

File: 0041-first-missing-positive/0041-first-missing-positive.java
class Solution {
    public int firstMissingPositive(int[] nums) {
        Set<Integer> pq=new HashSet<>();
        for(int i=0;i<nums.length;i++)
        {
            pq.add(nums[i]);
        }
        int a=1;
        while(pq.contains(a))
            
        {
            a++;
        }
        return a;
    }
}